Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -108,6 +108,16 @@ static inline std::span<char*> span(char** strv)
     return unsafeMakeSpan(strv, size);
 }
 
+static inline std::span<void*> span(GPtrArray* array)
+{
+    return unsafeMakeSpan(array->pdata, array->len);
+}
+
+static inline std::span<void*> span(GRefPtr<GPtrArray>& array)
+{
+    return span(array.get());
+}
```

Comment:
We could turn those into a template with the default type being `void*`, and then avoid manual casts in the rest of the code (see below). Something like:

```suggestion
template <typename T = void*, typename = std::enable_if_t<std::is_pointer_v<T>>>
inline std::span<T> span(GPtrArray* array)
{
    return unsafeMakeSpan(static_cast<T**>(static_cast<void*>(array ? array->pdata : nullptr)), array ? array->len : 0);
}

template <typename T = void*, typename = std::enable_if_t<std::is_pointer_v<T>>>
inline std::span<T> span(GRefPtr<GPtrArray>& array)
{
    return span<T>(array.get());
}
```

Reply:
Thanks, I did something like this with small variations.